src/directives/formly-field.js

```diff
@@ -152,13 +152,26 @@ function formlyField($http, $q, $compile, $templateCache, formlyConfig, formlyVa
     function resetModel() {
       $scope.model[$scope.options.key] = $scope.options.initialValue;
       if ($scope.options.formControl) {
-        $scope.options.formControl.$setViewValue($scope.model[$scope.options.key]);
-        $scope.options.formControl.$render();
-        $scope.options.formControl.$setUntouched();
-        $scope.options.formControl.$setPristine();
+        if(angular.isArray($scope.options.formControl)){
+          angular.forEach($scope.options.formControl, function(formControl){
+            resetFormControl(formControl, true);
+          });
+        } else {
+          resetFormControl($scope.options.formControl);
+        }
       }
     }
 
+    function resetFormControl(formControl, isMultiNgModel){
+      if(!isMultiNgModel){
+        formControl.$setViewValue($scope.model[$scope.options.key]);
+      }
```

Comment:
This would be a breaking change if we require digest to be called manually after resetFormControl.  While looking up ways to be safe with calling $digest it seems that checking the $$phase is not good form so I defer to you the decision about how to best handle this.